src/dodal/devices/i24/pmac.py

```diff
@@ -150,17 +154,19 @@ async def kickoff(self):
         )
 
     @AsyncStatus.wrap
-    async def complete(self):
-        """Stop collecting when the scan status PV goes to 0.
-
-        Args:
-            complete_time (float): total time required by the collection to \
-            finish correctly.
+    async def complete(self, counter_timeout: float = 30):
```

Comment:
Should: `complete` should implement the function as defined in `Flyable` and so not take any parameters. A better way to do this would be a soft signal that holds the timeout time